### ISSUE TITLE
Model overfitting on domain transfer - add preprocessing data augmentation (noise), dropout by default

### ISSUE THREAD

**Issue by breznak** [facebookresearch/detectron2#1890]
## 🚀 Feature

make models more stable by default, avoid overfitting. Either by data augmentation - adding noise - (in `DataLoader`?) or by using `dropout` layer, which I believe should help the problem too (but haven't implemented and validated)

## Motivation & Examples

Hello, 
my models overfit using default `train_net.py` with slight modification just to register custom dataset. 

This is due to no default data augmentation. I know this is common knowledge and users should take care, but at the time of getting detectron running at all, this was especially confusing since other frameworks (eg YOLACT) do that by default. That is aim of my feature proposal - do some sane data augmentation by default, so models do not overfit by default. 

This could be done either by: 
- mentioning this in "training your model" READMEs
- add noise (gauss, salt&pepper, speckle) as a preprocessing step (in `DataLoader`) 
- implement dropout at the layers, which should alleviate the problem (?, I haven't tested)

**Comment by ppwwyyxx** [comment]
> This is due to no default data augmentation

There is data augmentation in every config provided. 

The augmentation we used in configs follow the academic common practice and we're unlikely to change that. Users who need more augmentations should take care of adding more.
